### PR TITLE
Feat: Auto close logic for Inactive User, Added Turns and Timer for the Users (reopened)

### LINKED ISSUE #16: Add auto close logic
If the user is inactive for more than 60s
Probably needs to be configurable

### PR DESCRIPTION
**Pull request by Airbornharsh**
issue close #16 

1. Auto Close logic after 1 min timeout
2. Can see your and opponents turn
3. Timer in the frontend for you and for the Opponent

### CODE REVIEW

File: apps/ws/src/GameManager.ts [hunk 1]
```diff
@@ -60,6 +64,32 @@ export class GameManager {
                 }
             }
 
+            if (message.type === GAME_OVER){
+                const gameId = message.payload.gameId;
```
Comment (hkirat) [inline apps/ws/src/GameManager.ts:68]: code duplicated?

Comment (hkirat) [reply apps/ws/src/GameManager.ts:68]: Why is this needed?
When does this code execute?

**Comment by Airbornharsh** [comment]
@hkirat you can this now

**Comment by hkirat** [comment]
merge conflicts

**Comment by Airbornharsh** [comment]
@hkirat 

1. Auto Close logic after 1 min timeout
2. Can see your and opponents turn
3. Timer in the frontend for you and for the Opponent

https://github.com/code100x/chess/assets/95279849/c4cf83e0-a426-4646-a05d-69d7c4f324c3

**Comment by hkirat** [comment]
Amazing, will be checking this today

**Comment by Airbornharsh** [comment]
> Amazing, will be checking this today

Okay bhaiya
if need update in this, then comment

File: apps/ws/src/GameManager.ts [hunk 2]
```diff
@@ -85,6 +90,12 @@ export class GameManager {
         const game = this.games.find((game) => game.gameId === gameId);
         if (game) {
           game.makeMove(user, message.payload.move);
+          game.clearTimer();
+          const timer = setTimeout(() => {
+            game.endGame();
+            this.removeGame(game.gameId);
+          }, 60 * 1000);
```
Comment (hkirat) [inline apps/ws/src/GameManager.ts:97]: note for later, move this some place else this smells a bit
make time configurable

File: apps/ws/src/Game.ts [hunk 3]
```diff
@@ -151,6 +166,27 @@ export class Game {
...
+          type: USER_TIMEOUT,
+          payload: {
+            win: this.player1Time <= 0 ? 'BLACK_WINS' : 'WHITE_WINS',
+          },
+        }),
+      );
+      await db.game.update({
+        data: {
+          status: 'COMPLETED',
+          result: this.player1Time <= 0 ? 'BLACK_WINS' : 'WHITE_WINS',
```
Comment (hkirat) [inline apps/ws/src/Game.ts:182]: note for later - add reason here in db

File: apps/ws/src/Game.ts [hunk 4]
```diff
@@ -207,4 +256,43 @@ export class Game {
...
+  }
+
+  clearTimer() {
+    if (this.timer) clearTimeout(this.timer);
+  }
+
+  updateUserTimer(user: User) {
+    const time = new Date(Date.now()).getTime();
+    if (user.userId === this.player1UserId) {
+      this.player1Time -= time - this.tempTime;
```
Comment (hkirat) [inline apps/ws/src/Game.ts:292]: tempTime?

**Pull request merged**